twisted/test/test_factories.py

```diff
@@ -91,6 +92,34 @@ def connect(self):
         self.assertFalse(f.clock.getDelayedCalls())
 
 
+    def test_jitteredDelayIsNonNegative(self):
+        """
+        Reconnection delay should not be negative.
+        """
+        class FakeClock:
+            def callLater(self, _seconds, _f, *args, **kw):
+                pass
+
+        # Let's not mess with other tests involving randomness.
+        original = random.getstate()
```

Comment:
well, we should call `self.addCleanup(random.setstate, original)` just after this call... in case the tests fails the random state is not reverted.

.... but the best way to have clean tests is to allow injecting the `random` dependecy ... just like we inject the fake clock

```
class ReconnectingClientFactory:
    _random = random

    def someMethod(self):
        self.delay = max(self._random())
```